src/nvim/edit.c

```diff
@@ -5280,14 +5281,15 @@ insertchar (
 
       (*mb_char2bytes)(c, buf);
       buf[cc] = NUL;
-      ins_char_bytes(buf, cc);
+      ins_char_bytes(buf, cc, false);
       AppendCharToRedobuff(c);
     } else {
-      ins_char(c);
-      if (flags & INSCHAR_CTRLV)
+      ins_char(c, true);
```

Comment:
See the comment on line 5219. Whatever you change in `ins_char`, you need to do in `ins_str` as well.